### DISCUSSION TITLE
Constrain floating element

### DISCUSSION THREAD

**Discussion by jzbuchalski** [floating-ui/floating-ui#2181] [Q&A] [answered]
I would like to constrain a tooltip (floating) with a certain div.  Any ideas?

In this sandbox, I want the tooltip always constrained within the green div:
https://codesandbox.io/s/floating-ui-constraint-3u4icl

**Comment by atomiks** [comment, marked as answer]
You can use the [`detectOverflow#boundary` option](https://floating-ui.com/docs/detectOverflow#boundary) which is available for both `flip` and `shift`: https://codesandbox.io/s/floating-ui-constraint-forked-ruwk1l?file=/src/App.tsx
